src/java.base/windows/classes/sun/nio/fs/WindowsFileStore.java

```diff
@@ -229,15 +239,22 @@ public boolean supportsFileAttributeView(String name) {
     public boolean equals(Object ob) {
         if (ob == this)
             return true;
-        if (!(ob instanceof WindowsFileStore))
-            return false;
-        WindowsFileStore other = (WindowsFileStore)ob;
-        return root.equals(other.root);
+        if (ob instanceof WindowsFileStore other) {
+            if (root.equals(other.root))
+                return true;
+            if (volType == DRIVE_FIXED && other.volumeType() == DRIVE_FIXED)
+                return root.equalsIgnoreCase(other.root);
+        }
+        return false;
     }
 
     @Override
     public int hashCode() {
-        return root.hashCode();
+        if (hashCode == 0) { // Don't care about race
+            hashCode = (volType == DRIVE_FIXED) ?
+                root.toLowerCase(Locale.ROOT).hashCode() : root.hashCode();
+        }
+        return hashCode;
```

Comment:
As I know. to make data race benign, field should be read only once. But there is second read which can still return `0`

Reply:
You can are right, missed in the final review (as this one went through many iterations).